src/test/PoolSettleTest.sol

```diff
@@ -3,18 +3,22 @@ pragma solidity ^0.8.24;
 
 import {IPoolManager} from "../interfaces/IPoolManager.sol";
 import {PoolKey} from "../types/PoolKey.sol";
+import {Currency} from "../types/Currency.sol";
 import {PoolTestBase} from "./PoolTestBase.sol";
 
 contract PoolSettleTest is PoolTestBase {
     constructor(IPoolManager _manager) PoolTestBase(_manager) {}
 
-    function settle() external payable {
-        manager.unlock("");
+    function settle(Currency syncCurrency) external payable {
+        manager.unlock(abi.encode(syncCurrency));
     }
 
-    function unlockCallback(bytes calldata) external returns (bytes memory) {
+    function unlockCallback(bytes calldata data) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
+        Currency syncCurrency = abi.decode(data, (Currency));
+        manager.sync(syncCurrency);
```

Comment:
you shouldnt need to call sync here if the currency we are settling is native?

Reply:
Its only used for 1 test, and its a test where the synced currency is _different_ than whats being settled. So just had to move that inside this contract. But arguably given this contract is only used for 1 test we should just use actions router

Reply:
ok i deleted PoolSettleTest and rewrote it with actions router